### PR TITLE
Bump mermaid 10.6.0, marked 9.1.2, rjsf 5.13.2

### LINKED ISSUE #15309: CI failure: linux docs - due to warning emitted by sphinx template
## Description

Our CI generating the doc considers warning as error. An [upstream change](https://github.com/pydata/pydata-sphinx-theme/pull/1503) to the pydata sphinx template emits a warning that breaks the CI.

```
WARNING: The default value for `navigation_with_keys` will change to `False` in the next release. If you wish to preserve the old behavior for your site, set `navigation_with_keys=True` in the `html_theme_options` dict in your `conf.py` file.Be aware that `navigation_with_keys = True` has negative accessibility implications:https://github.com/pydata/pydata-sphinx-theme/issues/1492
```

### PR DESCRIPTION
**Pull request by bollwyvl**
## References

Fixes #15309

## Code changes

- [x] update js deps
  - [x] `mermaid 10.6.0` for bug fixes, performance, and new xychart
  - [x] `marked 9.1.2` for bug fixes
  - [x] `rjsf 5.13.2` for bug fixes
  - [ ] ???
- [ ] docs
  - [x] disable custom JS keyboard navigation in `pydata-sphinx-theme` (also raises warning)
  - [ ] ???
- [ ] ???

## User-facing changes

- large mermaid charts should render more reliably

## Backwards-incompatible changes

- n/a

### CODE REVIEW

**Comment by bollwyvl** [comment]
Welp, it ran good once: https://github.com/jupyterlab/jupyterlab/pull/15302/commits/cd7348ed556f1bc80e3d665926d549f1f2fb088e :sweat_smile:

**Comment by krassowski** [comment]
>  disable custom JS keyboard navigation in pydata-sphinx-theme (also raises warning)

We will want to backport this one, might be slightly easier if it is in a separate PR.

**Comment by bollwyvl** [comment]
@krassowski  backport up on https://github.com/jupyterlab/jupyterlab/pull/15304

File: docs/source/conf.py [hunk 1]
```diff
@@ -284,6 +284,8 @@ def document_plugins_tokens_list(list_path: Path, output_path: Path) -> None:
         ),
         "version_match": os.environ.get("READTHEDOCS_VERSION", "latest"),
     },
+    # rely on browser-native accessibility features instead of custom JS
+    "navigation_with_keys": False,
```
Comment (fcollonval) [inline docs/source/conf.py:288]: Removing as it is gonna end up twice in the configuration with the fix from #15310 

```suggestion
```

**Review by fcollonval — APPROVED** [review]
Thanks @bollwyvl 

CI failures are not related.

**Comment by bollwyvl** [comment]
Thanks all :hugs:

**Pull request merged**